### PR TITLE
Unification of the contents of TryScanToken and ScanTokenFullWidth.

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
+ Small Refactoring with the unified method.
  + Use a common lengthWithMeyBeEquals
  + Change section to use IF( , ) instead.

### CODE REVIEW

**Comment by AdamSpeight2008** [comment]
Also note that on my local machine I get the following single failure.
```  Microsoft.CodeAnalysis.UnitTests.Diagnostics.DiagnosticLocalizationTests.TestDiagnosticLocalization

Assert.Equal() Failure
Position: First difference is at position 0
Expected: My Resource 1 ENU string
Actual:   

   at Microsoft.CodeAnalysis.UnitTests.Diagnostics.DiagnosticLocalizationTests.TestDiagnosticLocalization() in Src\Compilers\Core\CodeAnalysisTest\Diagnostics\DiagnosticLocalizationTests.cs:line 70
```

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb [hunk 1]
```diff
@@ -974,17 +974,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
...
-        ' at this point it is very likely that we are located at 
-        ' the beginning of a token        
-        Private Function TryScanToken(precedingTrivia As SyntaxList(Of VisualBasicSyntaxNode)) As SyntaxToken
-
-            If Not CanGetChar() Then
-                Return MakeEofToken(precedingTrivia)
-            End If
-
-            Dim ch As Char = PeekChar()
+        Private Function _Unified_(precedingTrivia As SyntaxList(Of VisualBasicSyntaxNode), ch As Char, fullWidth As Boolean) As SyntaxToken
```
Comment (VladimirReshetnikov) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: Are you going to rename this method eventually?

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb [hunk 2]
```diff
@@ -974,17 +974,13 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
...
+            Dim lengthWithMaybeEquals = 1
             Select Case ch
-                Case CARRIAGE_RETURN, LINE_FEED, NEXT_LINE, LINE_SEPARATOR, PARAGRAPH_SEPARATOR
```
Comment (VladimirReshetnikov) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: Could you please explain with this case is now split?

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb [hunk 3]
```diff
@@ -1198,227 +1175,41 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
...
-                        Return XmlMakeEndEmbeddedToken(precedingTrivia, _scanSingleLineTriviaFunc)
-                    End If
+        ' at this point it is very likely that we are located at 
+        ' the beginning of a token        
+        Private Function TryScanToken(precedingTrivia As SyntaxList(Of VisualBasicSyntaxNode)) As SyntaxToken
+            If Not CanGetChar() Then
+                Return MakeEofToken(precedingTrivia)
+            End If
+            Dim ch As Char = PeekChar()
+            Return _Unified_(precedingTrivia,ch,False)
```
Comment (VladimirReshetnikov) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: Please use spaces after commas.

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb [hunk 4]
```diff
@@ -1198,227 +1175,41 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
...
-            If IsIdentifierStartCharacter(ch) Then
-                Return ScanIdentifierOrKeyword(precedingTrivia)
-            End If
 
-            Debug.Assert(Not IsNewLine(ch))
-            Debug.Assert(Not IsDoubleQuote(ch))
 
-            Return Nothing
+        Private Function ScanTokenFullWidth(precedingTrivia As SyntaxList(Of VisualBasicSyntaxNode), ch As Char) As SyntaxToken
+            Return _Unified_(precedingTrivia,ch,True)
```
Comment (VladimirReshetnikov) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: Please use spaces after commas.

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb (hunk 1 above)
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: I am open to suggestions for the name.

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb (hunk 2 above)
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: In the previous methods, one doesn't have all of the choices. Splitting it permits a condition  , whilst the other choice can still be used fof both.

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb (hunk 3 above)
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: @VladimirReshetnikov will do.

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb (hunk 4 above)
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: @VladimirReshetnikov will do.

File: src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb (hunk 1 above)
Comment (AdamSpeight2008) [inline src/Compilers/VisualBasic/Portable/Scanner/Scanner.vb]: @VladimirReshetnikov 
Current thoughts are on the lines of
1.  `ScanToken_Common`
2. `ScamTokenIternals`

**Comment by AdamSpeight2008** [comment]
My remote repo's branch was different to my local one, and the git commands I used screw this up.
I unable to reopen do this `The Scanner(Unification) was force pushed or recreated.`
Suggestions?

**Comment by jaredpar** [comment]
@AdamSpeight2008 the following set of commands will take the contents of `Scanner_Unify` (PR #1427) and move them to `Scanner(Unification)` (this PR).  Note:

- This set of commands assumes that the remote `origin` points to https://github.com/AdamSpeight2008/roslyn
- This **will overwrite** the contents of `Scanner(Unification)`.  If you want to keep this around you should save it in another branch before executing these commands. 

That being said here is the code to move the new code into this PR.  

First step is to switch the branch to `Scanner(Unification)` which is the basis of this PR. 

```
git checkout Scanner(Unification)
```

Next we want this branch to have the same contents as `Scanner_Unify`.  

```
git reset --hard Scanner_Unify
```

Now we want to push this up to GitHub so that this PR will update.

```
git push origin Scanner(Unification):Scanner(Unification) -force
```

The use of `-force` is necessary here because these branches don't have a shared history (at least that I can see).

**Comment by AdamSpeight2008** [comment]
result
```
error: dst ref refs/heads/Scanner(Unification) receives from more than one src.
error: failed to push some refs to 'https://github.com/AdamSpeight2008/roslyn-AdamSpeight2008.git'
```

**Comment by AdamSpeight2008** [comment]
@jaredpar I think I fix it now.

**Comment by AdamSpeight2008** [comment]
@jaredpar I been in contact with GitHub on how to fix this issue.

> Ivan Žužak <support@github.com>
> 
> Sorry, Adam -- we can't do anything about that pull request after you force pushed and deleted the  head branch once. You'll need to open a new pull request.

> Let us know if you have any other questions.
>
> Cheers,
> Ivan

Please consider #1463 as the fix. *I'll not touch that branch at all*

**Pull request closed without merging**